influxdb.js

```diff
@@ -135,6 +135,11 @@ module.exports = function(RED) {
                 }
 
                 client.writePoints(points, writeOptions).catch(function(err) {
+                    if (err.res === undefined){
```

Comment:
Rather than adding a statusCode to a message when an exception occurs, can we simply add the error to the message like this?

```
msg.error = err;
node.error(err, msg);
```

This way the user can dig through the ResultError or RequestError by accessing the _error field in the message.  Not sure it will work though.  We may have to copy the fields that we want into a new object.

Reply:
![image](https://user-images.githubusercontent.com/6776991/64278162-1c281900-cf1a-11e9-8161-184310cf8e2a.png)
I tried that, weirdly though I don't think it worked. 

Below in the 400 you can see that my new code has added it to the msg, but not the err (a sanity check to make sure the write was working) and then you can see in the 503 there is nothing at all. I'm not sure where this _error value is, but the msg.error value seems to be populated by err, it's just not clear how this happens.

Forced a 400 with a bad query:
```
object
_msgid: "df7d6275.1c17e"
topic: ""
payload: 1567619174488
statusCode: 400
error: object
message: "Error: A 400 Bad Request error occurred: {"error":"error parsing query: only ORDER BY time supported at this time"}↵"
source: object
id: "fe9d378a.fdd1e8"
type: "influxdb in"
name: "Test Direct"
count: 1
stack: "Error: A 400 Bad Request error occurred: {"error":"error parsing query: only ORDER BY time supported at this time"}↵↵ at IncomingMessage.Create.res.on (***\node-red\node_modules\influx\lib\src\pool.js:49:38)↵ at IncomingMessage.emit (events.js:203:15)↵ at endReadableNT (_stream_readable.js:1129:12)↵ at process._tickCallback (internal/process/next_tick.js:63:19)"
```

Forced a 503 by turning off wifi
```
object
_msgid: "c8d62b5d.403e18"
topic: ""
payload: 1567619289863
error: object
message: "Error: getaddrinfo ENOTFOUND ***"
source: object
id: "fe9d378a.fdd1e8"
type: "influxdb in"
name: "Test Direct"
count: 1
stack: "Error: getaddrinfo ENOTFOUND *****
at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:56:26)"
```

Reply:
Do you see what I mean? Is there something I'm not understanding about how to use the first argument of node.error maybe? Maybe my logging is wrong too, I'm actually logging the msg. However when I added a function and attempted to access the `_error` not sure if it's on the global context or something, I would have thought it would be in the `msg`, I got a reference error

Reply:
yeah, you can't really mess with the error parameter passed in to node.error().  Looking at the node-red code, it pulls out just the fields it wants/expects and copies them over.  see https://github.com/node-red/node-red/blob/afaf077aca517b1de15a2e3e95277853a0b64549/packages/node_modules/%40node-red/runtime/lib/nodes/flows/Flow.js#L486


If you set msg.error, it looks like that code copies that to msg._error.  See https://github.com/node-red/node-red/blob/afaf077aca517b1de15a2e3e95277853a0b64549/packages/node_modules/%40node-red/runtime/lib/nodes/flows/Flow.js#L483-L485

I tried it yesterday, and seemed to work, but didn't test thoroughly with the different error types (ResultError and RequestError) thrown by the library.

Reply:
If we do this in each catch block:

```
msg.error = {
  statusCode: err.res ? err.res.statusCode : 503
};
node.error(err, msg);
```

it works.  A field called `_error` appears in the message along with the `error` field in the message to the catch node.  We can add more fields to this if needed later.

Reply:
Looks alright to me. The only gripe I might have is that you would have to track that down to the Node-Red runtime (as you've done here) to see that the field being injected into `error` is now in `_error`. Is this a documented node-red pattern (i.e. like recommended in the help or anything like that)? Maybe we could use another name like `influx_error` or something?

Reply:
I think we'd need to document it either way.  I suspect that magical move from `error` to `_error` is to support the edge case where a message already has an error field... so you're right, we probably don't want to depend on that.  Sure - lets put the fields in `influx-error`.  Do you want to make the change here and test it out, document it in the README?

Reply:
Ok so it's kind of frustrating that they don't throw the same error in the read and the write functions but I've updated the code to be the same for all of them. Basically the read only throws an Error whereas the writes (from the influx client) throw the connection information as well. Anyway this basically results in the reads always being a `503` if they fail, and the writes will have error codes like 404 for database not available, 400 for bad write, and 503 for a refused connection from the database.

My use-case is pretty much uniquely for writes, so this doesn't matter to me, but it might impact your decision on if you want to merge.

```
{ Error: Error from InfluxDB: database not found: testmeasurement
    at assertNoErrors (C:\Users\mw00120\bidsolutions\node-red\node_modules\influx\lib\src\results.js:92:19)
    at Object.parse (C:\Users\mw00120\bidsolutions\node-red\node_modules\influx\lib\src\results.js:108:5)
    at queryRaw.then.res (C:\Users\mw00120\bidsolutions\node-red\node_modules\influx\lib\src\index.js:849:68)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  message: 'Error from InfluxDB: database not found: testmeasurement' }
279
{ statusCode: 503 }
```